File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/server/level/LevelManager.java
package pize.tests.voxelgame.server.level;

import pize.tests.voxelgame.server.Server;
import pize.tests.voxelgame.server.chunk.gen.ChunkGenerator;

import java.util.Collection;
import java.util.HashMap;
import java.util.Map;

public class LevelManager{
    
    public final Server server;
    public final Map<String, ServerLevel> loadedLevels;
    
    public LevelManager(Server server){
        this.server = server;
        
        loadedLevels = new HashMap<>();
    }
    
    public Server getServer(){
        return server;
    }
    
    
    public Collection<ServerLevel> getLoadedLevels(){
        return loadedLevels.values();
    }
    
    public ServerLevel getLevel(String worldName){
        return loadedLevels.get(worldName);
    }
    
    public ServerLevel getDefaultLevel(){
        return getLevel(server.getConfiguration().getDefaultLevelName());
    }
    
    public void loadLevel(String levelName){
        if(levelName == null || isLevelExists(levelName) || isLevelLoaded(levelName))
            return;
        
        // final ServerLevel level = new ServerLevel(server);
        // level.getConfiguration().load(levelName, seed, generator);
        // loadedLevels.put(levelName, level);
        // level.getChunkManager().start();
        
        System.out.println("[Server]: Loaded level '" + levelName + "'");
    }
    
    public void createLevel(String levelName, int seed, ChunkGenerator generator){
        if(levelName == null || !isLevelExists(levelName) || isLevelLoaded(levelName))
            return;
        
        final ServerLevel level = new ServerLevel(server);
        level.getConfiguration().load(levelName, seed, generator);
        loadedLevels.put(levelName, level);
        level.getChunkManager().start();
        
        System.out.println("[Server]: Created level '" + levelName + "'");
    }
    
    public boolean isLevelLoaded(String levelName){
        return loadedLevels.containsKey(levelName);
    }
    
    public boolean isLevelExists(String levelName){
        return true; //: SHO? SHIZA.
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/server/level/LevelManager.kt
package pize.tests.voxelgame.server.level

import pize.tests.voxelgame.server.Server
import pize.tests.voxelgame.server.chunk.gen.ChunkGenerator

class LevelManager(val server: Server) {
    val loadedLevels: MutableMap<String?, ServerLevel>

    init {
        loadedLevels = HashMap()
    }

    fun getLoadedLevels(): Collection<ServerLevel> {
        return loadedLevels.values
    }

    fun getLevel(worldName: String?): ServerLevel? {
        return loadedLevels[worldName]
    }

    val defaultLevel: ServerLevel?
        get() = getLevel(server.configuration.defaultLevelName)

    fun loadLevel(levelName: String?) {
        if (levelName == null || isLevelExists(levelName) || isLevelLoaded(levelName)) return

        // final ServerLevel level = new ServerLevel(server);
        // level.getConfiguration().load(levelName, seed, generator);
        // loadedLevels.put(levelName, level);
        // level.getChunkManager().start();
        println("[Server]: Loaded level '$levelName'")
    }

    fun createLevel(levelName: String?, seed: Int, generator: ChunkGenerator?) {
        if (levelName == null || !isLevelExists(levelName) || isLevelLoaded(levelName)) return
        val level = ServerLevel(server)
        level.configuration.load(levelName, seed, generator)
        loadedLevels[levelName] = level
        level.chunkManager.start()
        println("[Server]: Created level '$levelName'")
    }

    fun isLevelLoaded(levelName: String?): Boolean {
        return loadedLevels.containsKey(levelName)
    }

    fun isLevelExists(levelName: String?): Boolean {
        return true //: SHO? SHIZA.
    }
}
